### PR TITLE
Catch spotinst exceptions

### PR DESCRIPTION
**Pull request by kelibezhani**
- Added SpotinstException
- Fixed error handling

### CODE REVIEW

File: disco_aws_automation/disco_elastigroup.py [hunk 1]
```diff
@@ -97,6 +97,9 @@ def get_existing_groups(self, hostclass=None, group_name=None):
         try:
             groups = self._spotinst_call().json()['response']['items']
             groups = [group for group in groups if group['name'].startswith(self.environment_name)]
+        except SpotinstException as err:
+            logger.info('Unable to get existing Spotinst groups: %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Probably better to do a `logger.exception` here to preserve the stacktrace. Also you wouldn't need the `err.message` anymore.

File: disco_aws_automation/disco_elastigroup.py [hunk 2]
```diff
@@ -126,7 +129,11 @@ def get_existing_group(self, hostclass=None, group_name=None, throw_on_two_group
 
     def _get_group_instances(self, group_id):
         """Returns list of instance ids in a group"""
-        return self._spotinst_call(path='/' + group_id + '/status').json()['response']['items']
+        try:
+            return self._spotinst_call(path='/' + group_id + '/status').json()['response']['items']
+        except SpotinstException as err:
+            logger.info('Unable to get Spotinst group instances: %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Same.

File: disco_aws_automation/disco_elastigroup.py [hunk 3]
```diff
@@ -290,18 +297,27 @@ def update_group(self, hostclass, desired_size=None, min_size=None, max_size=Non
             # Remove fields not allowed in update
             del group_config['group']['capacity']['unit']
             del group_config['group']['compute']['product']
-            self._spotinst_call(path='/' + group_id, data=group_config, method='put')
-            return {'name': group['name']}
+            try:
+                self._spotinst_call(path='/' + group_id, data=group_config, method='put')
+                return {'name': group['name']}
+            except SpotinstException as err:
+                logger.info('Unable to update Spotinst group: %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Same.

File: disco_aws_automation/disco_elastigroup.py [hunk 4]
```diff
@@ -290,18 +297,27 @@ def update_group(self, hostclass, desired_size=None, min_size=None, max_size=Non
...
         else:
-            new_group = self._spotinst_call(data=group_config, method='post').json()
-            new_group_name = new_group['response']['items'][0]['name']
-            return {'name': new_group_name}
+            try:
+                new_group = self._spotinst_call(data=group_config, method='post').json()
+                new_group_name = new_group['response']['items'][0]['name']
+                return {'name': new_group_name}
+            except SpotinstException as err:
+                logger.info('Unable to create Spotinst group: %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Same.

File: disco_aws_automation/disco_elastigroup.py [hunk 5]
```diff
@@ -290,18 +297,27 @@ def update_group(self, hostclass, desired_size=None, min_size=None, max_size=Non
...
 
     def _delete_group(self, group_id, force=False):
         """Delete an elastigroup by group id"""
         # We need argument `force` to match method in autoscale
         # pylint: disable=unused-argument
-        self._spotinst_call(path='/' + group_id, method='delete')
+        try:
+            self._spotinst_call(path='/' + group_id, method='delete')
+        except SpotinstException as err:
+            logger.info('Unable to delete Spotinst group: %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Same.

File: disco_aws_automation/disco_elastigroup.py [hunk 6]
```diff
@@ -332,7 +348,10 @@ def scaledown_groups(self, hostclass=None, group_name=None, wait=False, noerror=
                 }
             }
             logger.info("Scaling down group %s", group['name'])
-            self._spotinst_call(path='/' + group['id'], data=group_update, method='put')
+            try:
+                self._spotinst_call(path='/' + group['id'], data=group_update, method='put')
+            except SpotinstException as err:
+                logger.info('Unable to scaledown Spotinst group(s): %s', err.message)
```
Comment (jconstance-amplify) [inline disco_aws_automation/disco_elastigroup.py]: Same.

File: disco_aws_automation/disco_aws.py [hunk 7]
```diff
@@ -79,7 +78,7 @@ def __init__(self, config, environment_name=None, boto2_conn=None, vpc=None, rem
         self._disco_storage = storage or None  # lazily initialized
         self._discogroup = discogroup or None  # lazily initialized
         self._autoscale = autoscale or None  # lazily initialized
-        self._elastigroup = elastigroup or None  # lazily initialized
+        # self._elastigroup = elastigroup or None  # lazily initialized
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_aws.py]: commented out code

Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_aws.py]: This function will no longer work https://github.com/amplifylitco/asiaq/blob/3be9f67113619bc78274fdcf3cc89d10c1163dd8/disco_aws_automation/disco_aws.py#L152

File: disco_aws_automation/disco_vpc.py [hunk 8]
```diff
@@ -578,9 +578,9 @@ def get_all_subnets(self):
     def _destroy_instances(self):
         """ Find all instances in vpc and terminate them """
         autoscale = DiscoAutoscale(environment_name=self.environment_name)
-        autoscale.delete_groups(force=True)
-        elastigroup = DiscoElastigroup(environment_name=self.environment_name)
-        elastigroup.delete_groups()
+        # autoscale.delete_groups(force=True)
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_vpc.py]: Commented out code

**Comment by jconstance-amplify** [comment]
Should update the rake test task to have an invalid spotinst api key at some point as well --> https://github.com/amplifylitco/asiaq/blob/master/tasks/test.rake#L20

**Pull request closed without merging**